File: hyein/week3/타겟넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(v, sum): 
        nonlocal answer
        if v == len(numbers):
            if sum == target:
                answer += 1
            return
        
        dfs(v+1, sum+numbers[v])
        dfs(v+1, sum-numbers[v])
        
    dfs(0, 0)
    return answer

from itertools import product
def solution2(numbers, target):
    l = [(x, -x) for x in numbers]
    s = list(map(sum, product(*l)))
    return s.count(target)

File: hyejee/week3/타겟넘버.swift
import Foundation

var count: Int = 0

func dfs(_ numbers: [Int], _ index: Int, _ currentTotal: Int, _ target: Int) {
    if index == numbers.count {
        if currentTotal == target {
            count += 1
        }
        return
    }

    dfs(numbers, index + 1, currentTotal + numbers[index], target)
    dfs(numbers, index + 1, currentTotal - numbers[index], target)
}

func solution(_ numbers:[Int], _ target:Int) -> Int {
    dfs(numbers, 0, 0, target)
    return count
}
